src/tokens/tokens.service.ts

```diff
@@ -81,18 +81,29 @@ export class TokensService {
 
   async getReceipt(id: string): Promise<EventStreamReply> {
     const response = await this.http
-      .get<EventStreamReply>(`${this.baseUrl}/reply/${id}`)
+      .get<EventStreamReply>(`${this.baseUrl}/reply/${id}`, {
+        validateStatus: status => {
+          return status < 300 || status === 404;
+        },
+      })
       .toPromise();
+    if (response.status === 404) {
+      throw new NotFoundException();
+    }
     return response.data;
   }
 
   async createPool(dto: TokenPool): Promise<AsyncResponse> {
+    const dataToPack = {
+      trackingId: dto.trackingId,
+      data: dto.data, // TODO: remove
```

Comment:
Not a blocker, but wondering why now isn't the time to do this removal?

Reply:
FireFly is currently packing things (like the transaction ID) into this blob. E2E tests and other usage of FireFly would break if I removed it now.